### PR TITLE
minErr: link to actual url and remove url from message params

### PR DESCRIPTION
**Pull request by Narretz**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?**

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our [guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [ ] Fix/Feature: [Docs](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#documentation) have been added/updated
- [ ] Fix/Feature: Tests have been added; existing tests pass

**Other information**:

### CODE REVIEW

File: src/minErr.js [hunk 1]
```diff
@@ -82,6 +82,11 @@ function isValidObjectMaxDepth(maxDepth) {
 
 function minErr(module, ErrorConstructor) {
   ErrorConstructor = ErrorConstructor || Error;
+
+  var url = 'https://code.angularjs.org/"NG_VERSION_FULL"/docs/errors/';
```
Comment (gkalpak) [inline src/minErr.js]: Why not use `https://errors.angularjs.org/...`?

File: src/minErr.js [hunk 2]
```diff
@@ -91,17 +96,22 @@ function minErr(module, ErrorConstructor) {
       }),
       paramPrefix, i;
 
+    // A minErr message has two parts: the message itself and the url that contains the
+    // encoded message.
+    // The message's parameters can contain other error messages which an also include error urls.
```
Comment (gkalpak) [inline src/minErr.js]: _an also_ --> _also_

File: src/minErr.js [hunk 3]
```diff
@@ -91,17 +96,22 @@ function minErr(module, ErrorConstructor) {
...
+    // To prevent the messages from getting to long, we strip the error urls from the parameters.
```
Comment (gkalpak) [inline src/minErr.js]: _to long_ --> _too long_

**Review by gkalpak — COMMENTED** [review]
Note that on minified scripts, `minErr` is replaced by a [more compact implementation][1] by `ng-closure-runner`. It is a good idea (imo) to keep the two in-sync.

##
BTW, I realized that our `vendor/` diretory only contains built `.jar` files (e.g. for [ng-closure-runner][2]). I think it would be nice to have the sources too (at least for `ng-closure-runner` which are really short), so that we can apply fixes/changes more easily.

[1]: https://github.com/angular/ng-closure-runner/blob/master/src/minErr.js
[2]: https://github.com/Narretz/angular.js/tree/4a75ff328afe77b37ef536efcedd59759ee5a398/vendor/ng-closure-runner

File: src/minErr.js (hunk 1 above)
Comment (Narretz) [reply src/minErr.js]: errors. ist just a redirect. Why not cut out the middle men?

**Comment by Narretz** [comment]
Could be good to have the source for the ng-closure-runner.
The minified version actually removes the message completely, leaving only the url - therefore this Pr shouldn't make a big difference

**Comment by Narretz** [comment]
Here's the difference between the current behavior and the new behavior, in Edge:

current (this is actually about twice as long)
edge-error-current

new (still long, but only one screen).
edge-error-new

This is still long, but it's much easier to extract a "working" url from this error.

Btw, it seem like the new Firefox version has improved the handling of really long errors - I can't even replicate the long error urls anymore.

File: src/minErr.js (hunk 1 above)
Comment (gkalpak) [reply src/minErr.js]: It is shorter and cleaner :grin:

I prefer the `errors.angularjs.org` URLs, but don't feel too strongly about it.
But I do think it is a good idea to have the same URLs in non-minified and minified versions, so if we change it here, I would like to see it changes in `ng-closure-runner` too 😇

**Comment by Narretz** [comment]
True, the minErr urls should match at least ... do you know where the MINERR_URL in ng-closure-runner is defined? It looks like the string is replaced before the script is run, same as we do in ng, but I can't find a reference to it: https://github.com/angular/ng-closure-runner/blob/e6521dad34706669c44947a4b9415965456f7635/src/minErr.js#L18

**Comment by Narretz** [comment]
Just noticed - if the url is fixed in the compiled minErr, we'd have to re-compile it for each new version to match, right? This doesn't seem practical.

**Comment by gkalpak** [comment]
Not sure what you mean by _"compiled minErr"_. `minErr` is replaced in the source by `ng-closure-runner` during minification (which happens as part of our build process).

It turns out the URL is [passed as a command-line argument][1].
So, we don't need to modify `ng-closure-runner` :tada: We pass it [here][2].

[1]: https://github.com/angular/ng-closure-runner/blob/f1c80b1d77a0cef786f75afc6ef494a3703d34df/src/org/angularjs/closurerunner/NgClosureRunner.java#L119
[2]: https://github.com/angular/angular.js/blob/98e0e047b0f705005b256c70feb4e6368ff3a591/lib/grunt/utils.js#L197

**Comment by Narretz** [comment]
> Not sure what you mean by "compiled minErr". 

I thought the version in the closure variant of minErr was set (and fixed) before we replace the minErr script.

I also remembered why I changed the error url: before Firebase, errors.angularjs wouldn't actually redirect to the specific version, it would just redirect to docs.angularjs. Since this is no longer the case, we don't have to change it to code.angularjs

**Comment by Narretz** [comment]
How does this look now @gkalpak?

File: src/minErr.js [hunk 4]
```diff
@@ -82,6 +82,11 @@ function isValidObjectMaxDepth(maxDepth) {
 
 function minErr(module, ErrorConstructor) {
   ErrorConstructor = ErrorConstructor || Error;
+
+  var url = 'https://errors.angularjs.org/"NG_VERSION_FULL"/';
+  var regex = url + '[\\s\\S]*';
```
Comment (gkalpak) [inline src/minErr.js]: Minor, but the RegExp will match any character in place of the `.`s that appear in the URL.

**Review by gkalpak — APPROVED** [review]

File: src/minErr.js (hunk 4 above)
Comment (Narretz) [reply src/minErr.js]: Good catch - I'll change it

**Pull request merged**